entity-framework/core/miscellaneous/cli/dbcontext-creation.md

```diff
@@ -4,30 +4,46 @@ author: bricelam
+database schema. In most cases, it is desirable that the `DbContext` thereby created is configured in a similar
+way to how it would be [configured at run time][2].
+
+There are various ways the tools try to create the `DbContext`:
 
 From application services
 -------------------------
 If your startup project is an ASP.NET Core app, the tools try to obtain the DbContext object from the application's
-service provider. They obtain it by invoking `Program.BuildWebHost()` and accessing the `IWebHost.Services` property. Any
-DbContext registered using `IServiceCollection.AddDbContext<TContext>()` can be found and created this way. This pattern
-was [introduced in ASP.NET Core 2.0][1]
+service provider.
+
+The tool first try to obtain the service provider by invoking `Program.BuildWebHost()` and accessing the
+`IWebHost.Services` property.
+
+> [!NOTE]
+> When you create a new ASP.NET Core 2.0 application, this hook is included by default. In previous versions of EF Core
+> and ASP.NET Core, the tools try to invoke `Startup.ConfigureServices` directly in order to obtain the application's
+> service provider, but this pattern no longer works correctly in ASP.NET Core 2.0 applications. If you are upgrading an
+> ASP.NET Core 1.x application to 2.0, you can[modify your `Program` class to follow the new pattern][3].
+
+The `DbContext` itself and any dependencies in its constructor need to be registered as services in the application's
+service provider. This can be easily achieved by taking an instance of `DbContextOptions<TContext>` as a
+[constructor argument][4] and using the [`AddDbContext<TContext>` method][5].
 
-Using the default constructor
------------------------------
-If the DbContext can't be obtained from the application service provider, the tools look for the DbContext type inside
-the project. They try to create it using its default constructor.
+Using a constructor with no parameters
+--------------------------------------
+If the DbContext can't be obtained from the application service provider, the tools look for the derived `DbContext`
+type inside the project. Then they try to create an instance using a constructor with no parameters. This can be the
+default constructor if the `DbContext` is configured using the [`OnConfiguring`][6] method.
 
 From a design-time factory
 --------------------------
-You can also tell the tools how to create your DbContext by implementing `IDesignTimeDbContextFactory`. If a class
-implementing this interface is found inside your project, the tools bypass the other ways of creating the DbContext.
-They always use the factory at design time. A factory is especially useful if you need to configure the DbContext
-differently for design time than at runtime.
+You can also tell the tools how to create your DbContext by implementing the `IDesignTimeDbContextFactory<TContext>`
+interface. If a class implementing this interface is found in the same project as the derived `DbContext`, the tools
```

Comment:
@bricelam is there still a restriction that the factory needs to be in the same project/assembly as the dervived `DbContext` type? I copied this from the original text in configuring-dbcontext.

Reply:
It can either be in the startup assembly or in the same assembly as the `DbContext`. Most of the time, you'll probably want it in the startup assembly since that is typically where you set up your IoC container.

Reply:
Updated.